applications/StructuralMechanicsApplication/custom_constitutive/constitutive_laws_integrators/generic_constitutive_law_integrator_plasticity.h

```diff
@@ -0,0 +1,657 @@
+    typedef TYieldSurfaceType YieldSurfaceType;
+
+    /// The type of plastic potential 
+    typedef typename YieldSurfaceType::PlasticPotentialType PlasticPotentialType;
+
+    /// Counted pointer of GenericConstitutiveLawIntegratorPlasticity
+    KRATOS_CLASS_POINTER_DEFINITION(GenericConstitutiveLawIntegratorPlasticity);
+
+    ///@}
+    ///@name  Enum's
+    ///@{
+
+    enum class HardeningCurveType {LinearSoftening = 0, ExponentialSoftening = 1, InitialHardeningExponentialSoftening = 2};
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Initialization constructor
+    GenericConstitutiveLawIntegratorPlasticity()
+    {
+    }
+
+    /// Copy constructor
+    GenericConstitutiveLawIntegratorPlasticity(GenericConstitutiveLawIntegratorPlasticity const& rOther)
+    {
+    }
+
+    /// Assignment operator
+    GenericConstitutiveLawIntegratorPlasticity& operator=(GenericConstitutiveLawIntegratorPlasticity const& rOther)
+    {
+        return *this;
+    }
+
+    /// Destructor
+    virtual ~GenericConstitutiveLawIntegratorPlasticity()
+    {
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+
+    /**
+     * @brief This method integrates the predictive stress vector with the CL using differents evolution laws using the backward euler scheme
+     * @param PredictiveStressVector The predictive stress vector S = C:(E-Ep)
+     * @param StrainVector The equivalent strain vector of that integration point
+     * @param UniaxialStress The equivalent uniaxial stress 
+     * @param Threshold The maximum uniaxial stress of the linear behaviour
+     * @param PlasticDenominator The plasticity numerical value to obtain the pastic consistency factor
+     * @param Fflux The derivative of the yield surface
+     * @param Gflux The derivative of the plastic potential
+     * @param PlasticDissipation The internal variable of energy dissipation due to plasticity
+     * @param PlasticStrainIncrement The increment of plastic strain of this time step
+     * @param C The elastic constitutive matrix
+     * @param PlasticStrain The elastic constitutive matrix
+     * @param rMaterialProperties The material properties
+     * @param CharacteristicLength The equivalent length of the FE
+     */
+    static void IntegrateStressVector(
+        Vector& rPredictiveStressVector,
+        Vector& StrainVector, 
+        double& UniaxialStress, 
+        double& Threshold, 
+        double& PlasticDenominator, 
+        Vector& Fflux, 
+        Vector& Gflux, 
+        double& PlasticDissipation, 
+        Vector& PlasticStrainIncrement,  
+        const Matrix& C, 
+        Vector& PlasticStrain, 
+        const Properties& rMaterialProperties,
+        const double CharacteristicLength
+    )
+    {
+        bool is_converged = false; 
+        int iteration = 0, max_iter = 9000;  
+        BoundedVector<double, 6> DSigma, DS;
+        double PlasticConsistencyFactorIncrement, F;  
```

Comment:
please don't use cryptic variable names like `F`, this makes it hard for others to understand the code